Tests/BackendIntegrationTests/BaseBackendIntegrationTests.swift

```diff
@@ -97,6 +98,10 @@ class BaseBackendIntegrationTests: TestCase {
 
         self.createUserDefaults()
 
+        self.documentsDirectory = URL
+            .cachesDirectory
+            .appendingPathComponent(UUID().uuidString, conformingTo: .directory)
+
```

Comment:
Moved initialization here like `UserDefaults` so the entire test run uses the same folder, even if calling `resetSingleton()`.